nessie/src/main/java/org/apache/iceberg/nessie/NessieIcebergClient.java

```diff
@@ -223,27 +284,57 @@ namespace, getRef().getName()),
   }
 
   public boolean dropNamespace(Namespace namespace) throws NamespaceNotEmptyException {
+
+    getRef().checkMutable();
+    ContentKey key = ContentKey.of(namespace.levels());
+
     try {
-      getRef().checkMutable();
-      withReference(
-              getApi()
-                  .deleteNamespace()
-                  .namespace(org.projectnessie.model.Namespace.of(namespace.levels())))
-          .delete();
-      refresh();
+
+      CommitMultipleOperationsBuilder commitBuilder =
+          api.commitMultipleOperations()
+              .commitMeta(NessieUtil.buildCommitMetadata("drop namespace " + key, catalogOptions))
+              .operation(Operation.Delete.of(key));
+
+      Tasks.foreach(commitBuilder)
+          .retry(5)
+          .stopRetryOn(
+              NessieReferenceNotFoundException.class, NessieReferenceConflictException.class)
+          .throwFailureWhenFinished()
+          .onFailure((o, exception) -> refresh())
+          .run(
+              b -> {
+                Branch branch = b.branch((Branch) getReference()).commit();
+                getRef().updateReference(branch);
+              },
+              BaseNessieClientServerException.class);
+
       return true;
-    } catch (NessieNamespaceNotFoundException e) {
-      return false;
-    } catch (NessieNotFoundException e) {
+    } catch (NessieReferenceNotFoundException e) {
       LOG.error(
           "Cannot drop Namespace '{}': ref '{}' is no longer valid.",
           namespace,
           getRef().getName(),
           e);
       return false;
-    } catch (NessieNamespaceNotEmptyException e) {
-      throw new NamespaceNotEmptyException(
-          e, "Namespace '%s' is not empty. One or more tables exist.", namespace);
+    } catch (NessieReferenceConflictException e) {
+      List<Conflict> conflicts = e.getErrorDetails().conflicts();
```

Comment:
Suggestion (also for the conflict handling above):

```
Optional<Conflict.ConflictType> conflict = e.getErrorDetails().conflicts().stream().filter(c -> key.equals(c.key())).map(Conflict::conflictType).findFirst();
if (conflict.isPresent()) {
  switch (conflict.get()) {
    case KEY_DOES_NOT_EXIST: return false;
    case NAMESPACE_NOT_EMPTY: throw ...;
  }
}
```

Maybe worth to handle all conflicts in one method used for create-ref and drop-ref?

Reply:
The issue is that `Conflict.conflictType()` is nullable, so the above code raises a few compiler warnings.

> Maybe worth to handle all conflicts in one method used for create-ref and drop-ref?

There is already a method to handle all conflicts: `NessieTableOperations.maybeThrowSpecializedException()`. I didn't want to touch it because it would require some refactoring to be usable here. But I will check again.